src/components/Calculator.js

```diff
@@ -1,43 +1,66 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      obj: {
+        total: null,
+        next: null,
+        operation: null,
+
+      },
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { obj } = this.state;
+    this.setState({
+      obj: calculate(obj, buttonName),
+    });
+  }
+
   render() {
+    const { obj } = this.state;
+
     return (
       <div className="calcMainContainer">
-        <div className="calcDisplay">0</div>
+        <div className="calcDisplay">{obj.next || obj.operation || obj.total || 0 }</div>
         <div className="calcKeysContainer">
           <div className="calcKeys">
             <div className="calcNumbers">
-              <button type="button">AC</button>
-              <button type="button">+/-</button>
-              <button type="button">%</button>
+              <button type="button" onClick={() => this.handleClick('AC')}>AC</button>
+              <button type="button" onClick={() => this.handleClick('+/-')}>+/-</button>
+              <button type="button" onClick={() => this.handleClick('%')}>%</button>
```

Comment:
- [ ] One of the project requirements is that props need to be passed from the parent component (`Calculator`) to a child component. I would advise you to make the buttons you display a component and pass props from the Calculator component to the Button component. You are free to implement this requirement however you deem fit, but Kindly pass props from the Calculator component to the child component. 👍